flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/KinesisDataFetcher.java

```diff
@@ -584,17 +594,34 @@ private static ShardMetricsReporter registerShardMetrics(MetricGroup metricGroup
 	//  Miscellaneous utility functions
 	// ------------------------------------------------------------------------
 
+	/**
+	 * Function to map a Kinesis shard to a Flink subtask index.
+	 */
+	public interface ShardToSubtaskIndexFn {
```

Comment:
I would prefer if this is a top-level class, instead of nested in `KinesisDataFetcher` (which is actually an internal class).

Also, the `Fn` name could also be polished a bit. Maybe something like `KinesisShardAssigner` would be better? That would also be coherent name-wise with the Kafka side, which has a `KafkaTopicPartitionAssigner` class (though it isn't exposed).

Reply:
This also needs to be `Serializable`, since it'll be shipped along with the `JobGraph` to the JM.